src/chart/graph/circularLayoutHelper.ts

```diff
@@ -163,11 +184,51 @@ const _layoutNodesBasedOn: Record<'value' | 'symbolSize', LayoutNode> = {
             const radianHalf = halfRemainRadian + _symbolRadiansHalf[node.dataIndex];
 
             angle += radianHalf;
-            node.setLayout([
+            // auto circular layout for not dragged node
+            !node.getFixed() && node.setLayout([
                 r * Math.cos(angle) + cx,
                 r * Math.sin(angle) + cy
             ]);
             angle += radianHalf;
         });
     }
 };
+
+export function rotateNodeLabel(
+    node: GraphNode,
+    circularRotateLabel: boolean,
+    cx: number,
+    cy: number
+) {
+    const nodeModel = node.getModel<GraphNodeItemOption>();
+    const el = node.getGraphicEl() as Symbol;
```

Comment:
We need to check if `el` exists. `'-'` value may not create node element.

Reply:
done🤣